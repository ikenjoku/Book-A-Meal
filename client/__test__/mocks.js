
export const meals = [
  {
    id: 0,
    name: 'Nigerian Jollof',
    description: 'Max Jollof rice, plantain, grilled turkey, bellefull Coke',
    price: 1100,
    imageurl: 'https://africa-publicmages/products/80/80418_1465475724_ma.jpg',
    createdAt: new Date(),
    updatedAt: new Date(),
  },
  {
    id: 1,
    name: 'Rice and Beans',
    description: 'Rice, Beans, Plantain, Panla Sauce, Max Coke',
    price: 1500,
    imageurl: 'https://africa-g/images/products/80/80418_1465475724_ma.jpg',
    createdAt: new Date(),
    updatedAt: new Date(),
  },
  {
    id: 2,
    name: 'Egusi Semo',
    description: 'Semolina, Egusi Soup, Meat, Extra',
    price: 1200,
    imageurl: 'https://africa-public.food.jcts/80/80418_1465475724_ma.jpg',
  },
  {
    id: 3,
    name: 'Green Calulu',
    description: 'Chicken salad with extra cabbage',
    price: 1250,
    imageurl: 'https://dynamic/production/ng/images/products/80/_ma.jpg',
    createdAt: new Date(),
    updatedAt: new Date(),
  },
  {
    id: 4,
    name: 'Onion Jappatti',
    description: 'Ingera palate with raw meat sauce',
    price: 2200,
    imageurl: 'https://dynamic/production/ng/images/produc475724_ma.jpg',
    createdAt: new Date(),
    updatedAt: new Date(),
  },
];

export const testMeal = {
  id: 3234,
  name: 'Strawberry flakes',
  description: 'Ingera palate with raw meat sauce',
  price: 2200,
  imageurl: 'https://dynamic/prodg/images/produc475724_ma.jpg',
  createdAt: new Date(),
  updatedAt: new Date(),
};

export const mockStoreData = {
  authReducer: {
    authLoading: true,
    user: {},
    isLoggedIn: false,
    error: null,
  },
  mealReducer: {
    meals: [
      {
        id: 3,
        name: 'Green Calulu',
        description: 'Chicken salad with extra cabbage',
        price: 1250,
        imageurl: 'https://dynamic/production/ng/images/products/80/_ma.jpg',
        createdAt: new Date(),
        updatedAt: new Date(),
      },
      {
        id: 4,
        name: 'Onion Jappatti',
        description: 'Ingera palate with raw meat sauce',
        price: 2200,
        imageurl: 'https://dynamic/production/ng/images/produc475724_ma.jpg',
        createdAt: new Date(),
        updatedAt: new Date(),
      },
    ],
    error: null,
    isLoading: false,

  },
  menuReducer: {
    selectedMenu: null,
    error: null,

  },
  setMenuReducer: {
    createdMenu: null,
    selectedDate: '',
    error: null,
    success: false,
    updatedMenu: null,
  },
  orderReducer: {
    orderedMeal: null,
    error: null,
    meals: [],
    orders: [],
    previousOrders: [],
    isLoadingOrders: false,
  },
};

export const createdMenu = {
  message: 'Menu for 2018-06-15 was successfully created',
  newMenu: {
    id: 24,
    date: '2018-06-15',
    updatedAt: '2018-07-02T17:37:46.023Z',
    createdAt: '2018-07-02T17:37:46.023Z',
  },
};

export const menus = [{
  id: 23,
  date: '2018-07-02',
  createdAt: '2018-07-01T23:56:23.508Z',
  updatedAt: '2018-07-01T23:56:23.508Z',
  Meals: [
    {
      id: 31,
      name: 'Beans and Bread',
      price: 800,
      description: 'Heavily loaded with smoked fish.',
      imageurl: 'https://res.cloudinary.com/ikeenjoku/image/upload/v1529133645/bookameal/2018-06-16T07:20:43.822ZVegetablerice%2Bplantain%2Bgoatmeat%2Bcoke.jpg.jpg',
      MealMenu: {
        mealId: 31,
        menuId: 23,
        createdAt: '2018-07-01T23:56:28.012Z',
        updatedAt: '2018-07-01T23:56:28.012Z',
      },
    },
    {
      id: 34,
      name: 'Fish balls soup',
      price: 1200,
      description: 'Deliciously light soup with white fish balls',
      imageurl: 'https://res.cloudinary.com/ikeenjoku/image/upload/v1529134583/bookameal/2018-06-16T07:36:22.300Zassortedmeat.jpg.jpg',
      MealMenu: {
        mealId: 34,
        menuId: 23,
        createdAt: '2018-07-01T23:56:32.313Z',
        updatedAt: '2018-07-01T23:56:32.313Z',
      },
    },
    {
      id: 30,
      name: 'Beef with veggie soup',
      price: 1150,
      description: 'Stir fried sliced beef with premium vegetables and spices',
      imageurl: 'https://res.cloudinary.com/ikeenjoku/image/upload/v1529136037/bookameal/2018-06-16T08:00:34.302ZUgwu-Archi.jpg.jpg',
      MealMenu: {
        mealId: 30,
        menuId: 23,
        createdAt: '2018-07-01T23:56:36.920Z',
        updatedAt: '2018-07-01T23:56:36.920Z',
      },
    },
  ],
},
{
  id: 24,
  date: new Date().toISOString().substr(0, 10),
  createdAt: '2018-07-01T23:56:23.508Z',
  updatedAt: '2018-07-01T23:56:23.508Z',
  Meals: [
    {
      id: 31,
      name: 'Beans and Bread',
      price: 800,
      description: 'Heavily loaded with smoked fish.',
      imageurl: 'https://res.cloudinary.com/ikeenjoku/image/upload/v1529133645/bookameal/2018-06-16T07:20:43.822ZVegetablerice%2Bplantain%2Bgoatmeat%2Bcoke.jpg.jpg',
      MealMenu: {
        mealId: 31,
        menuId: 23,
        createdAt: '2018-07-01T23:56:28.012Z',
        updatedAt: '2018-07-01T23:56:28.012Z',
      },
    },
    {
      id: 34,
      name: 'Fish balls soup',
      price: 1200,
      description: 'Deliciously light soup with white fish balls',
      imageurl: 'https://res.cloudinary.com/ikeenjoku/image/upload/v1529134583/bookameal/2018-06-16T07:36:22.300Zassortedmeat.jpg.jpg',
      MealMenu: {
        mealId: 34,
        menuId: 23,
        createdAt: '2018-07-01T23:56:32.313Z',
        updatedAt: '2018-07-01T23:56:32.313Z',
      },
    },
    {
      id: 30,
      name: 'Beef with veggie soup',
      price: 1150,
      description: 'Stir fried sliced beef with premium vegetables and spices',
      imageurl: 'https://res.cloudinary.com/ikeenjoku/image/upload/v1529136037/bookameal/2018-06-16T08:00:34.302ZUgwu-Archi.jpg.jpg',
      MealMenu: {
        mealId: 30,
        menuId: 23,
        createdAt: '2018-07-01T23:56:36.920Z',
        updatedAt: '2018-07-01T23:56:36.920Z',
      },
    },
  ],
}];

export const orders = [
  {
    id: 2,
    date: '2018-05-16',
    amount: 1500,
    userId: 1,
    mealId: 2,
    status: 'cancelled',
    createdAt: '2018-05-23T19:56:44.346Z',
    updatedAt: '2018-05-23T19:56:44.346Z',
    User: {
      id: 1,
      firstname: 'Ezinne',
      lastname: 'Njoku',
    },
    Meal: {
      id: 2,
      name: 'Rice and Beans',
      price: 1500,
    },
  },
  {
    id: 3,
    date: '2018-05-16',
    amount: 1200,
    userId: 2,
    mealId: 3,
    status: 'pending',
    createdAt: '2018-05-23T19:56:44.346Z',
    updatedAt: '2018-05-23T19:56:44.346Z',
    User: {
      id: 2,
      firstname: 'Kelechi',
      lastname: 'Njoku',
    },
    Meal: {
      id: 3,
      name: 'Egusi Semo',
      price: 1200,
    },
  },
  {
    id: 4,
    date: '2018-05-16',
    amount: 1500,
    userId: 3,
    mealId: 3,
    status: 'delivered',
    createdAt: '2018-05-23T19:56:44.346Z',
    updatedAt: '2018-05-23T19:56:44.346Z',
    User: {
      id: 3,
      firstname: 'Nnamdi',
      lastname: 'James',
    },
    Meal: {
      id: 3,
      name: 'Egusi Semo',
      price: 1200,
    },
  },
];
