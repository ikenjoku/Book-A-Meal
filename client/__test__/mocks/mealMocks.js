export const getAllMealsMock = [
  {
    name: 'test meal',
    price: 1230,
    description: 'this is a very nice meal',
    imageurl: 'https://sdfsdjhbsn/sdf.asp',
  },
];
