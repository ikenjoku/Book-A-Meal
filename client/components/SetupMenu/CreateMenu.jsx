import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import MealSetupForm from './MealSetupForm';
import { getAMeals, setAMenu } from '../../actions/setupMenuActions';

class CreateMenu extends Component {
  constructor(props) {
    super(props);
  }

  componentDidMount() {
    this.props.getAMeals({limit: 20, page: 1});
  }

  handleSubmit = (values) => {
    this.props.setAMenu(values)
    this.props.closeModal();
  }

  render() {
    const {
      meals, isCreating, isFetching,
    } = this.props;

    return (
      <div className='setup-form'>
        <h2>Create a menu</h2>
        {isFetching
          ?
          <h3>Fetching meals...</h3>
          :
          <MealSetupForm
            handleSubmit={this.handleSubmit}
            meals={meals}
            action="Create"
            isSubmitting={isCreating}
            date={this.props.selectedDate}
          />}
      </div>
    );
  }
}

CreateMenu.propTypes = {
  getAMeals: PropTypes.func.isRequired,
  setAMenu: PropTypes.func.isRequired,
  meals: PropTypes.arrayOf(PropTypes.object).isRequired,
  isFetching: PropTypes.bool.isRequired,
  isCreating: PropTypes.bool.isRequired,
  closeModal: PropTypes.func.isRequired,
  selectedDate: PropTypes.string.isRequired,
};

const mapStateToProps = state => ({
  meals: state.setupMenuReducer.meals,
  isFetching: state.setupMenuReducer.isFetching,
  isCreating: state.setupMenuReducer.isCreating,
  selectedDate: state.setupMenuReducer.selectedDate,
});

export default connect(mapStateToProps, { getAMeals, setAMenu })(CreateMenu);