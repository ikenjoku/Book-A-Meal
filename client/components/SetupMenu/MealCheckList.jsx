import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';


export const MealCheckList = ({ meals, handleSelect, isFetching }) => {
  return (isFetching ? 
    <h3>Fetching meals...</h3>
    :
    meals.map(meal => (
    <div key={meal.id}>
      <label>
        <input
          type="checkbox"
          name="meals"
          value={meal.id}
          onClick={handleSelect}
          />
          <span>{meal.name}</span>
      </label>
    </div>
  )));
};

export const mapStateToProps = state => ({
  isFetching: state.setupMenuReducer.isFetching,
});

MealCheckList.propTypes = {
  meals: PropTypes.arrayOf(PropTypes.object).isRequired,
  handleSelect: PropTypes.func.isRequired,
  isFetching: PropTypes.bool.isRequired,
};

export default connect(mapStateToProps, null)(MealCheckList);
