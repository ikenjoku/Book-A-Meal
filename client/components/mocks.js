

export const sampleOrders = {
  message: 'All orders retrieved successfully',
  orders: [
    {
      id: 2,
      date: '2018-05-16',
      amount: 1500,
      userId: 1,
      mealId: 2,
      status: 'cancelled',
      createdAt: '2018-05-23T19:56:44.346Z',
      updatedAt: '2018-05-23T19:56:44.346Z',
      User: {
        id: 1,
        firstname: 'Ezinne',
        lastname: 'Njoku',
      },
      Meal: {
        id: 2,
        name: 'Rice and Beans',
        price: 1500,
      },
    },
    {
      id: 3,
      date: '2018-05-16',
      amount: 1200,
      userId: 2,
      mealId: 3,
      status: 'pending',
      createdAt: '2018-05-23T19:56:44.346Z',
      updatedAt: '2018-05-23T19:56:44.346Z',
      User: {
        id: 2,
        firstname: 'Kelechi',
        lastname: 'Njoku',
      },
      Meal: {
        id: 3,
        name: 'Egusi Semo',
        price: 1200,
      },
    },
    {
      id: 4,
      date: '2018-05-16',
      amount: 1500,
      userId: 3,
      mealId: 3,
      status: 'delivered',
      createdAt: '2018-05-23T19:56:44.346Z',
      updatedAt: '2018-05-23T19:56:44.346Z',
      User: {
        id: 3,
        firstname: 'Nnamdi',
        lastname: 'James',
      },
      Meal: {
        id: 3,
        name: 'Egusi Semo',
        price: 1200,
      },
    },
    {
      id: 1,
      date: '2018-05-16',
      amount: 1100,
      userId: 1,
      mealId: 2,
      status: 'cancelled',
      createdAt: '2018-05-23T19:56:44.346Z',
      updatedAt: '2018-05-23T20:38:56.387Z',
      User: {
        id: 1,
        firstname: 'Ezinne',
        lastname: 'Njoku',
      },
      Meal: {
        id: 2,
        name: 'Rice and Beans',
        price: 1500,
      },
    },
    {
      id: 5,
      date: '2018-05-23',
      amount: 1500,
      userId: 2,
      mealId: 2,
      status: 'pending',
      createdAt: '2018-05-23T23:55:05.091Z',
      updatedAt: '2018-05-23T23:55:05.091Z',
      User: {
        id: 2,
        firstname: 'Kelechi',
        lastname: 'Njoku',
      },
      Meal: {
        id: 2,
        name: 'Rice and Beans',
        price: 1500,
      },
    },
    {
      id: 7,
      date: '2018-05-16',
      amount: 1200,
      userId: 2,
      mealId: 3,
      status: 'pending',
      createdAt: '2018-05-23T19:56:44.346Z',
      updatedAt: '2018-05-23T19:56:44.346Z',
      User: {
        id: 2,
        firstname: 'Kelechi',
        lastname: 'Njoku',
      },
      Meal: {
        id: 3,
        name: 'Egusi Semo',
        price: 1200,
      },
    },
    {
      id: 6,
      date: '2018-05-16',
      amount: 1500,
      userId: 3,
      mealId: 3,
      status: 'delivered',
      createdAt: '2018-05-23T19:56:44.346Z',
      updatedAt: '2018-05-23T19:56:44.346Z',
      User: {
        id: 3,
        firstname: 'Nnamdi',
        lastname: 'James',
      },
      Meal: {
        id: 3,
        name: 'Egusi Semo',
        price: 1200,
      },
    },
    {
      id: 8,
      date: '2018-05-16',
      amount: 1100,
      userId: 1,
      mealId: 2,
      status: 'cancelled',
      createdAt: '2018-05-23T19:56:44.346Z',
      updatedAt: '2018-05-23T20:38:56.387Z',
      User: {
        id: 1,
        firstname: 'Ezinne',
        lastname: 'Njoku',
      },
      Meal: {
        id: 2,
        name: 'Rice and Beans',
        price: 1500,
      },
    },
  ],
};

