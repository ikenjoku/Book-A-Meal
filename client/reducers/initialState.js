export default {
  authReducer: {
    authLoading: true,
    user: {},
    isLoggedIn: false,
    error: null,
  },
  mealReducer: {
    meals: [],
    error: null,
    isLoading: true,

  },
  menuReducer: {
    selectedMenu: null,
    error: null,

  },
  setMenuReducer: {
    createdMenu: null,
    selectedDate: '',
    error: null,
    success: false,
    updatedMenu: null,
  },
  orderReducer: {
    orderedMeal: null,
    error: null,
    meals: [],
    orders: [],
    previousOrders: [],
    isLoadingOrders:false,
  },
};
