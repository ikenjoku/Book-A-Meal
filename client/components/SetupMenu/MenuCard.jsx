import React, { Component } from "react";
import moment from 'moment';
import Modal from 'react-modal';
import { connect } from "react-redux";
import { getAMenu } from "../../actions/setupMenuActions";
import CreateMenu from './CreateMenu';
import UpdateMenu from './UpdateMenu';

class MenuCard extends Component{
  constructor(props){
    super(props)
    this.state = {
      isOpen: false,
    }
  }
  handleOpenModal = () => {
    this.setState({ isOpen: true });
  }

  handleCloseModal = () => {
    this.setState({ isOpen: false });
  }

  handleMenuUpdate() {
    this.handleOpenModal();
  }

  render(){

    const modalStyle = {
      overlay: {
        backgroundColor: 'rgba(0, 0, 0, 0.25)',
      },
      content: {
        top: '50%',
        left: '50%',
        right: 'auto',
        bottom: 'auto',
        marginRight: '-50%',
        transform: 'translate(-50%, -50%)',
        minWidth: '20rem',
        width: '60%',
        overflow: 'hidden',
      },
    };

    const renderedContent = this.props.currentMenu.date ?
        <div className='menu-card'>
        <h3>Menu for { moment(this.props.currentMenu.date).format("dddd, MMMM Do YYYY") }</h3>
          <div className='menu-card-meals'>
          <ul>
            {
              this.props.currentMenu.Meals.map(meal => <li>{ meal.name }</li>)
            }
          </ul>
          </div>
          <button
            onClick={this.handleOpenModal}
          >Update Menu</button>
          <Modal
            isOpen={this.state.isOpen}
            contentLabel="Menu"
            style={modalStyle}
          >
            <button
              value="&times;"
              onClick={this.handleCloseModal}
              className="close"
            >close</button>
             <UpdateMenu closeModal={this.handleCloseModal} />
          </Modal>
        </div> :
          <div className='no-menu-card'>
            <h3>No Menu has been set for this day</h3>
            <button
              onClick={this.handleOpenModal}
            >Create Menu</button>
            <div>
              <Modal
              isOpen={this.state.isOpen}
              contentLabel="Menu"
              style={modalStyle}
            >
              <button
                value="&times;"
                onClick={this.handleCloseModal}
                className="close"
              >close</button>
                <CreateMenu closeModal={this.handleCloseModal} />
            </Modal>
            </div>
          </div> 
    return(
      <div className=''>
        {renderedContent}
      </div>
    );
  }
}

const mapStateToProps = state => {
  return {
    currentMenu: state.setupMenuReducer.currentMenu,
    isLoadingMenu: state.setupMenuReducer.isLoadingMenu,
    error: state.setupMenuReducer.error
  }
}

export default connect(mapStateToProps, { getAMenu })(MenuCard);