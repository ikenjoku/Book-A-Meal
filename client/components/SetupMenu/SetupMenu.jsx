import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from "react-redux";
import DatePicker from '../DatePicker.jsx';
import MenuCard from './MenuCard.jsx';
import { getAMenu, setSelectedDate } from "../../actions/setupMenuActions";

export class SetupMenu extends Component {
  constructor(props) {
    super(props);

  }
  componentDidMount() {
    const today = new Date().toISOString().substr(0, 10);
    this.props.getAMenu({ selectedDate: today });
    this.props.setSelectedDate({ selectedDate: today });
  }

  render() {
    return (
      <main id='setup-menu-container'>
        <div className="cool-lg-text">
          <h2>Setup Menu</h2>
        </div>
        <DatePicker btnName='Find Menu'
          onSubmit={selectedDate => this.props.getAMenu(selectedDate)}
          onChange={selectedDate => this.props.setSelectedDate(selectedDate)}
        />
        <MenuCard />
      </main>
    );
  }
}

export default connect(null, { getAMenu, setSelectedDate })(SetupMenu);
