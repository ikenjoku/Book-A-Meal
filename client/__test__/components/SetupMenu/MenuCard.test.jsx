import React from 'react';
import { shallow } from 'enzyme';
import { MenuCard ,mapStateToProps } from '../../../components/SetupMenu/MenuCard';
import initialState from '../../../reducers/initialState'
import { menus } from '../../mocks';

describe('MenuCard Component', () => {
  let wrapper, getAMenu, isLoadingMenu, currentMenu, selectedDate;
  getAMenu = jest.fn();
  isLoadingMenu = false;
  selectedDate = '2018-07-02';
  currentMenu = menus[0]

  test('should render MenuCard correctly', () => {
    wrapper = shallow(
      <MenuCard
      isLoadingMenu={isLoadingMenu}
      getAMenu={getAMenu}
      currentMenu={currentMenu}
      selectedDate={selectedDate}
      />);
    expect(wrapper).toMatchSnapshot();
  });

  test('should render a notice card when no menu is set', () => {
    wrapper = shallow(
      <MenuCard
      isLoadingMenu={isLoadingMenu}
      getAMenu={getAMenu}
      currentMenu={{}}
      selectedDate={selectedDate}
      />);
      expect(wrapper.find('.no-menu-card').length).toBe(1);
  });

  test('should render a loader while fetching menu', () => {
    wrapper = shallow(
      <MenuCard
      isLoadingMenu={true}
      getAMenu={getAMenu}
      currentMenu={currentMenu}
      selectedDate={selectedDate}
      />);
      expect(wrapper.find('.loading-spinner').length).toBe(1);
  });

  it('should call handleOpenModal to open modal', () => {
    wrapper = shallow(
      <MenuCard
      isLoadingMenu={isLoadingMenu}
      getAMenu={getAMenu}
      currentMenu={currentMenu}
      selectedDate={selectedDate}
      />);

    const handleOpenModalSpy = jest.spyOn(
      wrapper.instance(), 'handleOpenModal'
    );
    wrapper.instance().handleOpenModal();
    expect(handleOpenModalSpy).toHaveBeenCalledTimes(1);
  });

  it('should call handleCloseModal to close modal', () => {
    wrapper = shallow(
      <MenuCard
      isLoadingMenu={isLoadingMenu}
      getAMenu={getAMenu}
      currentMenu={currentMenu}
      selectedDate={selectedDate}
      />);

    const handleCloseModalSpy = jest.spyOn(
      wrapper.instance(), 'handleCloseModal'
    );
    wrapper.instance().handleCloseModal();
    expect(handleCloseModalSpy).toHaveBeenCalledTimes(1);
  });

  test('should map state to props', () => {
    const ownProps = { 
      isLoadingMenu: false,
      selectedDate: '2018-07-02',
      currentMenu: menus[0],
    };
    const tree = mapStateToProps(initialState, ownProps);
    expect(tree).toMatchSnapshot();
  });
});
