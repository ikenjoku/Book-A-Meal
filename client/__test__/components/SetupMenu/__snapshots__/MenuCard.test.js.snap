// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuCard Component should map state to props 1`] = `
Object {
  "currentMenu": Object {},
  "isLoadingMenu": false,
  "selectedDate": "",
}
`;

exports[`MenuCard Component should render MenuCard correctly 1`] = `
<div
  className=""
>
  <div
    className="menu-card"
  >
    <h3>
      Menu for 
      Monday, July 2nd 2018
    </h3>
    <div
      className="menu-card-meals"
    >
      <ul>
        <li
          key="31"
        >
          Beans and Bread
        </li>
        <li
          key="34"
        >
          Fish balls soup
        </li>
        <li
          key="30"
        >
          Beef with veggie soup
        </li>
      </ul>
    </div>
    <button
      onClick={[Function]}
    >
      Update Menu
    </button>
    <Modal
      ariaHideApp={false}
      bodyOpenClassName="ReactModal__Body--open"
      closeTimeoutMS={0}
      contentLabel="Menu"
      isOpen={false}
      parentSelector={[Function]}
      portalClassName="ReactModalPortal"
      shouldCloseOnEsc={true}
      shouldCloseOnOverlayClick={true}
      shouldFocusAfterRender={true}
      shouldReturnFocusAfterClose={true}
      style={
        Object {
          "content": Object {
            "bottom": "auto",
            "left": "50%",
            "marginRight": "-50%",
            "minWidth": "20rem",
            "overflow": "hidden",
            "right": "auto",
            "top": "50%",
            "transform": "translate(-50%, -50%)",
            "width": "40%",
          },
          "overlay": Object {
            "backgroundColor": "rgba(0, 0, 0, 0.25)",
          },
        }
      }
    >
      <div
        className="close-icon"
      >
         
        <button
          onClick={[Function]}
        >
          <i
            className="fas fa-times fa-2x"
          />
        </button>
      </div>
      <Connect(UpdateMenu)
        closeModal={[Function]}
      />
    </Modal>
  </div>
</div>
`;
